### PR TITLE
Project Submission

### PR DESCRIPTION
**Pull request by preciousuwhubetine**
# Parks Plaza: A React Web Application for Exploring US Parks

## Overview
Parks Plaza is a web application that provides a directory of parks in the United States based on user-selected activities. The application is built with React and styled using SCSS, and is responsive for mobile devices.

## Features
Parks Plaza offers the following features:

### Home Page
- Activities route (`/`): This route displays a list of all available activities for users to select from.

### Park Directory
- Parks route (`/parks/:id`): This route displays a list of parks that offer the selected activity, identified by `id`.

## Styling
Parks Plaza follows a mobile-first approach to styling, with SCSS used for customization. While the site is responsive, it may not display optimally on large screens.

## Testing
Parks Plaza was rigorously tested using the React testing library. This included snapshot testing for the Parks and Activities components, as well as API call testing. To ensure seamless testing, a helper function from the Official Redux documentation was used to mock the redux store (`src/tests/components/utits.js`).

## Project Presentation Video
Watch me presenting the features of the project [here](https://www.loom.com/share/26fa1e2bb78b4216ba12e21d674d74ee).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,146 @@
...
+### Key Features <a name="key-features"></a>
+
+- A homepage displaying all available activities
+- A page displaying all the parks available for a specific activity
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## 🚀 Live Demo <a name="live-demo"></a>
+
+The project is live at [this](https://microverse-parks-plaza.vercel.app/) link.
```
Comment (franclobo) [inline README.md:51]: - [ ] Unfortunately, the deployment is not working as expected, please 🙏🏽 try to use Netlify 👇🏽 

image

File: src/tests/routes/activities.test.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+/**
+ * @jest-environment jsdom
+ */
```
Comment (franclobo) [inline src/tests/routes/activities.test.js:3]: - [ ] Please 🙏🏽 remove commented code while it is not to help to understand your code, remember YAGNI.

File: src/tests/routes/parks.test.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+/**
+ * @jest-environment jsdom
+ */
```
Comment (franclobo) [inline src/tests/routes/parks.test.js:3]: - [ ] Please 🙏🏽 remove commented code while it is not to help to understand your code, remember YAGNI.

**Review by franclobo — COMMENTED** [review]
Hi @preciousbetine, 

so-close-jim-halpert

Good job so far!
You are doing it well 👏🏽 there are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

## Highlights

- No linter errors ✔️ 
- Descriptive PR ✔️  
- Good UX ✔️ 
- Good UI ✔️ 

## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [x] Well done showing your app, but there are some requirements to make the video for professional documentation, please 🙏🏽 complete them all 👇🏽 

> a description of the project. ✅ 
> a demo of the project features. ✅ 
> you should also highlight some interesting piece of code or something you built that you are very proud of. ❌ 

- [x] Please 🙏🏽 create some integration tests using `@testing-library/react` to complete the tests requirements 👇🏽 

> Create integration tests for your application using the React Testing Library.
> You may need to mock the access to the API, so that your tests don't send actual requests.
> You may need to mock the connection to the Redux Store.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: README.md (hunk 1 above)
Comment (preciousuwhubetine) [reply README.md:51]: @franclobo, Thanks for the review. However, I respectfully want to point out that there are a few changes you suggested that may not be necessary:

First of all, I carried out integration testing with the react testing library. In the `tests/utils` file, I export a utility function that renders a component that has access to the redux store. This helper function is available from the official redux site.

I made use of the `render` function from react testing library here:

https://github.com/preciousbetine/microverse-parks-plaza/blob/cab1846ba2409482a91d95447aae27320d7a1834/src/tests/utils.js#L1-L3

Also, the deployed app is working. The link in my README displays the live site once I merge this branch to the dev branch. Vercel deployed this branch to preview earlier on in this PR thread.  The preview link is [here](https://microverse-parks-plaza-bze79q77i-preciousbetine.vercel.app/).

File: src/tests/routes/activities.test.js (hunk 2 above)
Comment (preciousuwhubetine) [reply src/tests/routes/activities.test.js:3]: @franclobo, thanks once again  but this comment block is required by jest to know the environment we're running tests in. In this case, the DOM environment

**Review by joskalenda — CHANGES_REQUESTED** [review]
| STATUS: REQUIRED CHANGES ♻️  |
|--------------------------------|
| Hi @preciousbetine , |
|Great work on making the changes requested by a previous reviewer 👏, You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet. |
|  |

## Highlights

- [ ].. ✅ Readme file
- [ ].. ✅ Good design
- [ ]...✅ Green linters

## Required Changes ♻️

- [ ] Hey @preciousbetine, you have done great work, but seeing the shot below, some of your test cases are not passing, it is required to have all the tests green before the project gets approved, please fix this and you will be good to go



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._


Cheers, and Happy coding!👏👏👏

-------
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me(@joskalenda) in your question so I can receive the notification.

> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by preciousuwhubetine** [comment]
@joskalenda Thanks for your review. However, on my system all test cases seem to be passing. In the screenshot you posted, the errors seem to have something to do with the version of node installed on your system. Here is a screenshot of all the tests passing on my system:

image

**Review by omar-muhamad — APPROVED** [review]
# Approved ✔
### Hi @preciousbetine,

#### Great job, Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

### Congratulations! 🎉 

giphy (1)

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- N/A

## [Optional] suggestions:

- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by preciousuwhubetine** [comment]
@Omar-Muhamad Thank you :)

**Pull request merged**